install ecmwflibs for climetlab fails on supercomputer

When I try to install `climetlab` and `climetlab-s2s-ai-competition` with most recent `pip` on a clean environment, I get the following:
Looks like `pip` tries to install all previous versions also. Is there a better way to install?

```bash
m300524@mistralpp4:~/s2s-ai-competition-bootstrap$ conda create --name test python=3.7 xarray netcdf4
m300524@mistralpp4:~/s2s-ai-competition-bootstrap$ source activate test
(/work/mh0727/m300524/conda-envs/test) m300524@mistralpp4:~/s2s-ai-competition-bootstrap$ pip install climetlab climetlab_s2s_ai_competition
```
<details>

Collecting climetlab
  Using cached climetlab-0.3.18.tar.gz (63 kB)
Collecting climetlab_s2s_ai_competition
  Using cached climetlab-s2s-ai-competition-0.3.6.tar.gz (6.1 kB)
Collecting cython
  Using cached Cython-0.29.22-cp37-cp37m-manylinux1_x86_64.whl (2.0 MB)
Requirement already satisfied: pandas in /mnt/lustre01/work/mh0727/m300524/conda-envs/test/lib/python3.7/site-packages (from climetlab) (1.2.3)
Collecting requests
  Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)
     |████████████████████████████████| 61 kB 448 kB/s
Requirement already satisfied: numpy in /mnt/lustre01/work/mh0727/m300524/conda-envs/test/lib/python3.7/site-packages (from climetlab) (1.20.1)
Requirement already satisfied: xarray>=0.17.0 in /mnt/lustre01/work/mh0727/m300524/conda-envs/test/lib/python3.7/site-packages (from climetlab) (0.17.0)
Requirement already satisfied: netcdf4 in /mnt/lustre01/work/mh0727/m300524/conda-envs/test/lib/python3.7/site-packages (from climetlab) (1.5.6)
Collecting cfgrib>=0.9.8.4
  Downloading cfgrib-0.9.8.5-py2.py3-none-any.whl (50 kB)
     |████████████████████████████████| 50 kB 376 kB/s
Collecting cdsapi
  Using cached cdsapi-0.5.1.tar.gz (12 kB)
Collecting ecmwf-api-client>=1.6.1
  Using cached ecmwf-api-client-1.6.1.tar.gz (11 kB)
Collecting tqdm
  Using cached tqdm-4.59.0-py2.py3-none-any.whl (74 kB)
Collecting eccodes>=0.9.9
  Using cached eccodes-1.2.0-py3-none-any.whl (46 kB)
Collecting magics>=1.5.6
  Using cached Magics-1.5.6.tar.gz (20 kB)
Collecting climetlab
  Using cached climetlab-0.3.16.tar.gz (62 kB)
  Using cached climetlab-0.3.15.tar.gz (62 kB)
  Using cached climetlab-0.3.14.tar.gz (62 kB)
  Using cached climetlab-0.3.13.tar.gz (62 kB)
  Using cached climetlab-0.3.11.tar.gz (61 kB)
  Using cached climetlab-0.3.10.tar.gz (61 kB)
  Using cached climetlab-0.3.8.tar.gz (61 kB)
  Using cached climetlab-0.3.7.tar.gz (63 kB)
  Using cached climetlab-0.3.6.tar.gz (62 kB)
  Using cached climetlab-0.3.5.tar.gz (62 kB)
  Using cached climetlab-0.3.4.tar.gz (62 kB)
  Using cached climetlab-0.3.3.tar.gz (62 kB)
  Using cached climetlab-0.3.0.tar.gz (62 kB)
  Using cached climetlab-0.2.7.tar.gz (60 kB)
  Using cached climetlab-0.2.4.tar.gz (58 kB)
  Using cached climetlab-0.2.3.tar.gz (58 kB)
Collecting chardet<4.0,>=3.0
  Using cached chardet-3.0.4-py2.py3-none-any.whl (133 kB)
Collecting aiohttp==3.7.2
  Using cached aiohttp-3.7.2.tar.gz (1.1 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting climetlab
  Using cached climetlab-0.1.10.tar.gz (52 kB)
  Using cached climetlab-0.1.7.tar.gz (52 kB)
  Using cached climetlab-0.1.6.tar.gz (52 kB)
  Using cached climetlab-0.1.5.tar.gz (52 kB)
  Using cached climetlab-0.1.2.tar.gz (45 kB)
  Using cached climetlab-0.1.0.tar.gz (45 kB)
Collecting pandas==1.0.5
  Using cached pandas-1.0.5-cp37-cp37m-manylinux1_x86_64.whl (10.1 MB)
Collecting climetlab
  Using cached climetlab-0.0.97.tar.gz (45 kB)
  Using cached climetlab-0.0.92.tar.gz (48 kB)
  Using cached climetlab-0.0.86.tar.gz (34 kB)
  Using cached climetlab-0.0.85.tar.gz (34 kB)
  Using cached climetlab-0.0.84.tar.gz (33 kB)
  Using cached climetlab-0.0.83.tar.gz (33 kB)
  Using cached climetlab-0.0.82.tar.gz (33 kB)
  Using cached climetlab-0.0.80.tar.gz (33 kB)
  Using cached climetlab-0.0.77.tar.gz (33 kB)
  Using cached climetlab-0.0.76.tar.gz (33 kB)
  Using cached climetlab-0.0.75.tar.gz (33 kB)
  Using cached climetlab-0.0.73.tar.gz (31 kB)
  Using cached climetlab-0.0.72.tar.gz (31 kB)
  Using cached climetlab-0.0.70.tar.gz (31 kB)
  Using cached climetlab-0.0.69.tar.gz (31 kB)
  Using cached climetlab-0.0.68.tar.gz (31 kB)
  Using cached climetlab-0.0.67.tar.gz (31 kB)
  Using cached climetlab-0.0.66.tar.gz (31 kB)
  Using cached climetlab-0.0.63.tar.gz (29 kB)
  Using cached climetlab-0.0.61.tar.gz (29 kB)
  Using cached climetlab-0.0.60.tar.gz (29 kB)
  Using cached climetlab-0.0.58.tar.gz (28 kB)
  Using cached climetlab-0.0.57.tar.gz (28 kB)
  Using cached climetlab-0.0.55.tar.gz (28 kB)
  Using cached climetlab-0.0.54.tar.gz (28 kB)
  Using cached climetlab-0.0.53.tar.gz (28 kB)
  Using cached climetlab-0.0.52.tar.gz (28 kB)
  Using cached climetlab-0.0.51.tar.gz (28 kB)
  Using cached climetlab-0.0.50.tar.gz (27 kB)
  Using cached climetlab-0.0.49.tar.gz (27 kB)
  Using cached climetlab-0.0.48.tar.gz (27 kB)
  Using cached climetlab-0.0.47.tar.gz (27 kB)
  Using cached climetlab-0.0.45.tar.gz (26 kB)
  Using cached climetlab-0.0.44.tar.gz (26 kB)
  Using cached climetlab-0.0.42.tar.gz (26 kB)
  Using cached climetlab-0.0.40.tar.gz (26 kB)
  Using cached climetlab-0.0.39.tar.gz (26 kB)
  Using cached climetlab-0.0.37.tar.gz (26 kB)
  Using cached climetlab-0.0.36.tar.gz (26 kB)
  Using cached climetlab-0.0.35.tar.gz (26 kB)
  Using cached climetlab-0.0.34.tar.gz (26 kB)
  Using cached climetlab-0.0.33.tar.gz (26 kB)
  Using cached climetlab-0.0.32.tar.gz (26 kB)
  Using cached climetlab-0.0.31.tar.gz (26 kB)
  Using cached climetlab-0.0.30.tar.gz (24 kB)
  Using cached climetlab-0.0.28.tar.gz (23 kB)
  Using cached climetlab-0.0.27.tar.gz (23 kB)
  Using cached climetlab-0.0.26.tar.gz (21 kB)
  Using cached climetlab-0.0.25.tar.gz (21 kB)
  Using cached climetlab-0.0.23.tar.gz (21 kB)
  Using cached climetlab-0.0.22.tar.gz (21 kB)
  Using cached climetlab-0.0.21.tar.gz (20 kB)
  Using cached climetlab-0.0.20.tar.gz (18 kB)
  Using cached climetlab-0.0.19.tar.gz (18 kB)
  Using cached climetlab-0.0.17.tar.gz (18 kB)
ERROR: Cannot install climetlab==0.0.17, climetlab==0.0.19, climetlab==0.0.20, climetlab==0.0.21, climetlab==0.0.22, climetlab==0.0.23, climetlab==0.0.25, climetlab==0.0.26, climetlab==0.0.27, climetlab==0.0.28, climetlab==0.0.30, climetlab==0.0.31, climetlab==0.0.32, climetlab==0.0.33, climetlab==0.0.34, climetlab==0.0.35, climetlab==0.0.36, climetlab==0.0.37, climetlab==0.0.39, climetlab==0.0.40, climetlab==0.0.42, climetlab==0.0.44, climetlab==0.0.45, climetlab==0.0.47, climetlab==0.0.48, climetlab==0.0.49, climetlab==0.0.50, climetlab==0.0.51, climetlab==0.0.52, climetlab==0.0.53, climetlab==0.0.54, climetlab==0.0.55, climetlab==0.0.57, climetlab==0.0.58, climetlab==0.0.60, climetlab==0.0.61, climetlab==0.0.63, climetlab==0.0.66, climetlab==0.0.67, climetlab==0.0.68, climetlab==0.0.69, climetlab==0.0.70, climetlab==0.0.72, climetlab==0.0.73, climetlab==0.0.75, climetlab==0.0.76, climetlab==0.0.77, climetlab==0.0.80, climetlab==0.0.82, climetlab==0.0.83, climetlab==0.0.84, climetlab==0.0.85, climetlab==0.0.86, climetlab==0.0.92, climetlab==0.0.97, climetlab==0.1.0, climetlab==0.1.10, climetlab==0.1.2, climetlab==0.1.5, climetlab==0.1.6, climetlab==0.1.7, climetlab==0.2.3, climetlab==0.2.4, climetlab==0.2.7, climetlab==0.3.0, climetlab==0.3.10, climetlab==0.3.11, climetlab==0.3.13, climetlab==0.3.14, climetlab==0.3.15, climetlab==0.3.16, climetlab==0.3.18, climetlab==0.3.3, climetlab==0.3.4, climetlab==0.3.5, climetlab==0.3.6, climetlab==0.3.7 and climetlab==0.3.8 because these package versions have conflicting dependencies.

The conflict is caused by:
    climetlab 0.3.18 depends on ecmwflibs>=0.1.2
    climetlab 0.3.16 depends on ecmwflibs>=0.0.91
    climetlab 0.3.15 depends on ecmwflibs>=0.0.91
    climetlab 0.3.14 depends on ecmwflibs>=0.0.91
    climetlab 0.3.13 depends on ecmwflibs>=0.0.91
    climetlab 0.3.11 depends on ecmwflibs>=0.0.91
    climetlab 0.3.10 depends on ecmwflibs>=0.0.91
    climetlab 0.3.8 depends on ecmwflibs>=0.0.91
    climetlab 0.3.7 depends on ecmwflibs>=0.0.91
    climetlab 0.3.6 depends on ecmwflibs>=0.0.91
    climetlab 0.3.5 depends on ecmwflibs>=0.0.91
    climetlab 0.3.4 depends on ecmwflibs>=0.0.91
    climetlab 0.3.3 depends on ecmwflibs>=0.0.91
    climetlab 0.3.0 depends on ecmwflibs>=0.0.91
    climetlab 0.2.7 depends on ecmwflibs>=0.0.91
    climetlab 0.2.4 depends on ecmwflibs>=0.0.91
    climetlab 0.2.3 depends on ecmwflibs>=0.0.91
    climetlab 0.1.10 depends on ecmwflibs>=0.0.91
    climetlab 0.1.7 depends on ecmwflibs>=0.0.91
    climetlab 0.1.6 depends on ecmwflibs>=0.0.91
    climetlab 0.1.5 depends on ecmwflibs>=0.0.91
    climetlab 0.1.2 depends on ecmwflibs>=0.0.90
    climetlab 0.1.0 depends on ecmwflibs>=0.0.88
    climetlab 0.0.97 depends on ecmwflibs>=0.0.39
    climetlab 0.0.92 depends on ecmwflibs
    climetlab 0.0.86 depends on ecmwflibs
    climetlab 0.0.85 depends on ecmwflibs
    climetlab 0.0.84 depends on ecmwflibs
    climetlab 0.0.83 depends on ecmwflibs
    climetlab 0.0.82 depends on ecmwflibs
    climetlab 0.0.80 depends on ecmwflibs
    climetlab 0.0.77 depends on ecmwflibs
    climetlab 0.0.76 depends on ecmwflibs
    climetlab 0.0.75 depends on ecmwflibs
    climetlab 0.0.73 depends on ecmwflibs
    climetlab 0.0.72 depends on ecmwflibs
    climetlab 0.0.70 depends on ecmwflibs
    climetlab 0.0.69 depends on ecmwflibs
    climetlab 0.0.68 depends on ecmwflibs
    climetlab 0.0.67 depends on ecmwflibs
    climetlab 0.0.66 depends on ecmwflibs
    climetlab 0.0.63 depends on ecmwflibs
    climetlab 0.0.61 depends on ecmwflibs
    climetlab 0.0.60 depends on ecmwflibs
    climetlab 0.0.58 depends on ecmwflibs
    climetlab 0.0.57 depends on ecmwflibs
    climetlab 0.0.55 depends on ecmwflibs
    climetlab 0.0.54 depends on ecmwflibs
    climetlab 0.0.53 depends on ecmwflibs
    climetlab 0.0.52 depends on ecmwflibs
    climetlab 0.0.51 depends on ecmwflibs
    climetlab 0.0.50 depends on ecmwflibs
    climetlab 0.0.49 depends on ecmwflibs
    climetlab 0.0.48 depends on ecmwflibs
    climetlab 0.0.47 depends on ecmwflibs
    climetlab 0.0.45 depends on ecmwflibs
    climetlab 0.0.44 depends on ecmwflibs
    climetlab 0.0.42 depends on ecmwflibs
    climetlab 0.0.40 depends on ecmwflibs
    climetlab 0.0.39 depends on ecmwflibs
    climetlab 0.0.37 depends on ecmwflibs
    climetlab 0.0.36 depends on ecmwflibs
    climetlab 0.0.35 depends on ecmwflibs
    climetlab 0.0.34 depends on ecmwflibs
    climetlab 0.0.33 depends on ecmwflibs
    climetlab 0.0.32 depends on ecmwflibs
    climetlab 0.0.31 depends on ecmwflibs
    climetlab 0.0.30 depends on ecmwflibs
    climetlab 0.0.28 depends on ecmwflibs
    climetlab 0.0.27 depends on ecmwflibs
    climetlab 0.0.26 depends on ecmwflibs
    climetlab 0.0.25 depends on ecmwflibs
    climetlab 0.0.23 depends on ecmwflibs
    climetlab 0.0.22 depends on ecmwflibs
    climetlab 0.0.21 depends on ecmwflibs
    climetlab 0.0.20 depends on ecmwflibs
    climetlab 0.0.19 depends on ecmwflibs
    climetlab 0.0.17 depends on ecmwflibs

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
(/work/mh0727/m300524/conda-envs/test) m300524@mistralpp4:~/s2s-ai-competition-bootstrap

</details>

`conda list`

<details>

# packages in environment at /work/mh0727/m300524/conda-envs/test:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
branca                    0.3.1                    pypi_0    pypi
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.17.1               h7f98852_1    conda-forge
ca-certificates           2020.12.5            ha878542_0    conda-forge
certifi                   2020.12.5        py37h89c1867_1    conda-forge
cftime                    1.4.1            py37h902c9e0_0    conda-forge
curl                      7.75.0               h979ede3_0    conda-forge
dask                      2021.3.0                 pypi_0    pypi
entrypoints               0.3                      pypi_0    pypi
folium                    0.12.1                   pypi_0    pypi
hdf4                      4.2.13            h10796ff_1004    conda-forge
hdf5                      1.10.6          nompi_h6a2412b_1114    conda-forge
jpeg                      9d                   h36c2ea0_0    conda-forge
krb5                      1.17.2               h926e7f8_0    conda-forge
ld_impl_linux-64          2.35.1               hea4e1c9_2    conda-forge
libblas                   3.9.0                8_openblas    conda-forge
libcblas                  3.9.0                8_openblas    conda-forge
libcurl                   7.75.0               hc4aaa36_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.3                  h58526e2_2    conda-forge
libgcc-ng                 9.3.0               h2828fa1_18    conda-forge
libgfortran-ng            9.3.0               hff62375_18    conda-forge
libgfortran5              9.3.0               hff62375_18    conda-forge
libgomp                   9.3.0               h2828fa1_18    conda-forge
liblapack                 3.9.0                8_openblas    conda-forge
libnetcdf                 4.7.4           nompi_h56d31a8_107    conda-forge
libnghttp2                1.43.0               h812cca2_0    conda-forge
libopenblas               0.3.12          pthreads_h4812303_1    conda-forge
libssh2                   1.9.0                ha56f1ee_6    conda-forge
libstdcxx-ng              9.3.0               h6de172a_18    conda-forge
markdown                  3.3.4                    pypi_0    pypi
ncurses                   6.2                  h58526e2_4    conda-forge
netcdf4                   1.5.6           nompi_py37hdbee05a_100    conda-forge
numpy                     1.20.1           py37haa41c4c_0    conda-forge
openssl                   1.1.1j               h7f98852_0    conda-forge
pandas                    1.2.3            py37hdc94413_0    conda-forge
pip                       21.0.1             pyhd8ed1ab_0    conda-forge
pyodc                     1.0.3                    pypi_0    pypi
python                    3.7.10          hffdb5ce_100_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2021.1             pyhd8ed1ab_0    conda-forge
pyyaml                    5.4.1                    pypi_0    pypi
readline                  8.0                  he28a2e2_2    conda-forge
setuptools                49.6.0           py37h89c1867_3    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sqlite                    3.35.2               h74cdb3f_0    conda-forge
tk                        8.6.10               h21135ba_1    conda-forge
toolz                     0.11.1                   pypi_0    pypi
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
xarray                    0.17.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
zlib                      1.2.11            h516909a_1010    conda-forge

</details>

`pip freeze`

<details>

branca==0.3.1
certifi==2020.12.5
cftime @ file:///home/conda/feedstock_root/build_artifacts/cftime_1612246430970/work
climetlab==0.4.4
dask==2021.3.0
entrypoints==0.3
folium==0.12.1
Markdown==3.3.4
netCDF4 @ file:///home/conda/feedstock_root/build_artifacts/netcdf4_1613286964717/work
numpy @ file:///home/conda/feedstock_root/build_artifacts/numpy_1612952999509/work
pandas==1.2.3
pdbufr==0.8.2
pyodc==1.0.3
python-dateutil==2.8.1
pytz @ file:///home/conda/feedstock_root/build_artifacts/pytz_1612179539967/work
PyYAML==5.4.1
six @ file:///home/conda/feedstock_root/build_artifacts/six_1590081179328/work
toolz==0.11.1
xarray @ file:///home/conda/feedstock_root/build_artifacts/xarray_1614346479764/work

</details>

on the supercomputer, when I try to install `ecmwflibs` from source:

`(/work/mh0727/m300524/conda-envs/test_climetlab) m300524@mistralpp4:~$ pip install git+https://github.com/ecmwf/ecmwflibs.git`

```bash
Collecting git+https://github.com/ecmwf/ecmwflibs.git
  Cloning https://github.com/ecmwf/ecmwflibs.git to /tmp/pip-req-build-uh272ibg
  Running command git clone -q https://github.com/ecmwf/ecmwflibs.git /tmp/pip-req-build-uh272ibg
Building wheels for collected packages: ecmwflibs
  Building wheel for ecmwflibs (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /work/mh0727/m300524/conda-envs/test_climetlab/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-uh272ibg/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-uh272ibg/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-ly38iwi4
       cwd: /tmp/pip-req-build-uh272ibg/
  Complete output (45 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.8
  creating build/lib.linux-x86_64-3.8/ecmwflibs
  copying ecmwflibs/__main__.py -> build/lib.linux-x86_64-3.8/ecmwflibs
  copying ecmwflibs/__init__.py -> build/lib.linux-x86_64-3.8/ecmwflibs
  running egg_info
  creating ecmwflibs.egg-info
  writing ecmwflibs.egg-info/PKG-INFO
  writing dependency_links to ecmwflibs.egg-info/dependency_links.txt
  writing top-level names to ecmwflibs.egg-info/top_level.txt
  writing manifest file 'ecmwflibs.egg-info/SOURCES.txt'
  reading manifest file 'ecmwflibs.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching 'climetlab/*/*.yaml'
  warning: no files found matching 'climetlab/*/*/*.yaml'
  warning: no files found matching 'climetlab/*/*/*/*.yaml'
  warning: no files found matching 'climetlab/*/*/*/*/*.yaml'
  warning: no files found matching 'climetlab/*/*/*/*/*/*.yaml'
  writing manifest file 'ecmwflibs.egg-info/SOURCES.txt'
  copying ecmwflibs/_ecmwflibs.cc -> build/lib.linux-x86_64-3.8/ecmwflibs
  creating build/lib.linux-x86_64-3.8/ecmwflibs/etc
  copying ecmwflibs/etc/README -> build/lib.linux-x86_64-3.8/ecmwflibs/etc
  creating build/lib.linux-x86_64-3.8/ecmwflibs/copying
  copying ecmwflibs/copying/.gitignore -> build/lib.linux-x86_64-3.8/ecmwflibs/copying
  running build_ext
  building 'ecmwflibs._ecmwflibs' extension
  creating build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/ecmwflibs
  gcc -pthread -B /work/mh0727/m300524/conda-envs/test_climetlab/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/tmp/pip-req-build-uh272ibg/install/include -I/tmp/pip-req-build-uh272ibg/install/include/magics -I/work/mh0727/m300524/conda-envs/test_climetlab/include/python3.8 -c ecmwflibs/_ecmwflibs.cc -o build/temp.linux-x86_64-3.8/ecmwflibs/_ecmwflibs.o
  cc1plus: warning: command line option "-Wstrict-prototypes" is valid for Ada/C/ObjC but not for C++
  ecmwflibs/_ecmwflibs.cc:4:21: error: eccodes.h: No such file or directory
  ecmwflibs/_ecmwflibs.cc:5:27: error: magics_config.h: No such file or directory
  ecmwflibs/_ecmwflibs.cc: In function ‘PyObject* versions(PyObject*, PyObject*)’:
  ecmwflibs/_ecmwflibs.cc:10: error: ‘grib_get_api_version’ was not declared in this scope
  ecmwflibs/_ecmwflibs.cc:11: warning: ‘auto’ will change meaning in C++0x; please remove it
  ecmwflibs/_ecmwflibs.cc:11: error: ISO C++ forbids declaration of ‘p’ with no type
  ecmwflibs/_ecmwflibs.cc:11: error: invalid conversion from ‘const char*’ to ‘int’
  ecmwflibs/_ecmwflibs.cc:15: error: ‘ECCODES_VERSION_STR’ was not declared in this scope
  ecmwflibs/_ecmwflibs.cc:16: error: ‘MAGICS_VERSION_STR’ was not declared in this scope
  ecmwflibs/_ecmwflibs.cc:10: warning: unused variable ‘s’
  ecmwflibs/_ecmwflibs.cc:11: warning: unused variable ‘p’
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for ecmwflibs
  Running setup.py clean for ecmwflibs
Failed to build ecmwflibs
Installing collected packages: ecmwflibs
    Running setup.py install for ecmwflibs ... error
    ERROR: Command errored out with exit status 1:
     command: /work/mh0727/m300524/conda-envs/test_climetlab/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-uh272ibg/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-uh272ibg/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-9cyazmd5/install-record.txt --single-version-externally-managed --compile --install-headers /work/mh0727/m300524/conda-envs/test_climetlab/include/python3.8/ecmwflibs
         cwd: /tmp/pip-req-build-uh272ibg/
    Complete output (43 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.8
    creating build/lib.linux-x86_64-3.8/ecmwflibs
    copying ecmwflibs/__main__.py -> build/lib.linux-x86_64-3.8/ecmwflibs
    copying ecmwflibs/__init__.py -> build/lib.linux-x86_64-3.8/ecmwflibs
    running egg_info
    writing ecmwflibs.egg-info/PKG-INFO
    writing dependency_links to ecmwflibs.egg-info/dependency_links.txt
    writing top-level names to ecmwflibs.egg-info/top_level.txt
    reading manifest file 'ecmwflibs.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching 'climetlab/*/*.yaml'
    warning: no files found matching 'climetlab/*/*/*.yaml'
    warning: no files found matching 'climetlab/*/*/*/*.yaml'
    warning: no files found matching 'climetlab/*/*/*/*/*.yaml'
    warning: no files found matching 'climetlab/*/*/*/*/*/*.yaml'
    writing manifest file 'ecmwflibs.egg-info/SOURCES.txt'
    copying ecmwflibs/_ecmwflibs.cc -> build/lib.linux-x86_64-3.8/ecmwflibs
    creating build/lib.linux-x86_64-3.8/ecmwflibs/etc
    copying ecmwflibs/etc/README -> build/lib.linux-x86_64-3.8/ecmwflibs/etc
    creating build/lib.linux-x86_64-3.8/ecmwflibs/copying
    copying ecmwflibs/copying/.gitignore -> build/lib.linux-x86_64-3.8/ecmwflibs/copying
    running build_ext
    building 'ecmwflibs._ecmwflibs' extension
    creating build/temp.linux-x86_64-3.8
    creating build/temp.linux-x86_64-3.8/ecmwflibs
    gcc -pthread -B /work/mh0727/m300524/conda-envs/test_climetlab/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/tmp/pip-req-build-uh272ibg/install/include -I/tmp/pip-req-build-uh272ibg/install/include/magics -I/work/mh0727/m300524/conda-envs/test_climetlab/include/python3.8 -c ecmwflibs/_ecmwflibs.cc -o build/temp.linux-x86_64-3.8/ecmwflibs/_ecmwflibs.o
    cc1plus: warning: command line option "-Wstrict-prototypes" is valid for Ada/C/ObjC but not for C++
    ecmwflibs/_ecmwflibs.cc:4:21: error: eccodes.h: No such file or directory
    ecmwflibs/_ecmwflibs.cc:5:27: error: magics_config.h: No such file or directory
    ecmwflibs/_ecmwflibs.cc: In function ‘PyObject* versions(PyObject*, PyObject*)’:
    ecmwflibs/_ecmwflibs.cc:10: error: ‘grib_get_api_version’ was not declared in this scope
    ecmwflibs/_ecmwflibs.cc:11: warning: ‘auto’ will change meaning in C++0x; please remove it
    ecmwflibs/_ecmwflibs.cc:11: error: ISO C++ forbids declaration of ‘p’ with no type
    ecmwflibs/_ecmwflibs.cc:11: error: invalid conversion from ‘const char*’ to ‘int’
    ecmwflibs/_ecmwflibs.cc:15: error: ‘ECCODES_VERSION_STR’ was not declared in this scope
    ecmwflibs/_ecmwflibs.cc:16: error: ‘MAGICS_VERSION_STR’ was not declared in this scope
    ecmwflibs/_ecmwflibs.cc:10: warning: unused variable ‘s’
    ecmwflibs/_ecmwflibs.cc:11: warning: unused variable ‘p’
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /work/mh0727/m300524/conda-envs/test_climetlab/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-uh272ibg/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-uh272ibg/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-9cyazmd5/install-record.txt --single-version-externally-managed --compile --install-headers /work/mh0727/m300524/conda-envs/test_climetlab/include/python3.8/ecmwflibs Check the logs for full command output.
```

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -27,4 +27,3 @@ jobs:
         pip install -e .
     - name: import climetlab_s2s_ai_competition
       run: |
-        which python

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -21,8 +21,7 @@ jobs:
         pip install -e .
         python -c "import climetlab_s2s_ai_competition"
 
-jobs_conda:
-  install-climetlab_s2s_ai_competition:
+  install-conda-climetlab_s2s_ai_competition:
     steps:
     - uses: actions/checkout@v2
     - name: Setup python

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -24,4 +24,3 @@ jobs:
         python -c "import climetlab_s2s_ai_competition"
       run: |
         python -m pip install --upgrade pip
-        pip install -e .

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -40,7 +40,7 @@ jobs:
     - name: Set up conda environment
       run: |
           conda create --name climetlab_s2s_ai_competition-dev python=3.7 xarray
-          source activate climetlab_s2s_ai_competition-dev
+          conda activate climetlab_s2s_ai_competition-dev
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -22,22 +22,6 @@ jobs:
     - name: import climetlab_s2s_ai_competition
       run: |
         python -c "import climetlab_s2s_ai_competition"
-
-  install-conda-climetlab_s2s_ai_competition:
-    name: install-conda-climetlab_s2s_ai_competition
-    runs-on: "ubuntu-latest"
-    defaults:
-      run:
-        shell: bash -l {0}
-    steps:
-    - uses: actions/checkout@v2
-    - name: Setup python
-      uses: actions/setup-python@v2
-      with:
-        python-version: 3.7
-    - name: Set up conda
-      uses: conda-incubator/setup-miniconda@v2
-    - name: Set up conda environment
       run: |
           conda create --name climetlab_s2s_ai_competition-dev python=3.7 xarray
           conda activate climetlab_s2s_ai_competition-dev

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -25,5 +25,4 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install -e .
-    - name: import climetlab_s2s_ai_competition
       run: |

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -30,9 +30,9 @@ jobs_conda:
       with:
         python-version: 3.7
     - name: Set up conda
-        uses: conda-incubator/setup-miniconda@v2
+      uses: conda-incubator/setup-miniconda@v2
     - name: Set up conda environment
-        run: |
+      run: |
           conda create --name climetlab_s2s_ai_competition-dev python=3.7 xarray
     - name: Install dependencies
       run: |

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -22,4 +22,3 @@ jobs:
     - name: import climetlab_s2s_ai_competition
       run: |
         python -c "import climetlab_s2s_ai_competition"
-        python -m pip install --upgrade pip

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -20,3 +20,22 @@ jobs:
         python -m pip install --upgrade pip
         pip install -e .
         python -c "import climetlab_s2s_ai_competition"
+
+jobs_conda:
+  install-climetlab_s2s_ai_competition:
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Set up conda
+        uses: conda-incubator/setup-miniconda@v2
+    - name: Set up conda environment
+        run: |
+          conda create --name climetlab_s2s_ai_competition-dev python=3.7 xarray
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .
+        python -c "import climetlab_s2s_ai_competition"

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -22,7 +22,6 @@ jobs:
     - name: import climetlab_s2s_ai_competition
       run: |
         python -c "import climetlab_s2s_ai_competition"
-      run: |
           conda create --name climetlab_s2s_ai_competition-dev python=3.7 xarray
           conda activate climetlab_s2s_ai_competition-dev
     - name: Install dependencies

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -25,4 +25,3 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install -e .
-      run: |

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -1,0 +1,22 @@
+name: climetlab_s2s_ai_competition installs
+
+on: pull_request
+
+jobs:
+  install-climetlab_s2s_ai_competition:  # Installs climetlab_s2s_ai_competition on various OS.
+    name: Install climetlab_s2s_ai_competition, ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .
+        python -c "import climetlab_s2s_ai_competition"

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -22,6 +22,11 @@ jobs:
         python -c "import climetlab_s2s_ai_competition"
 
   install-conda-climetlab_s2s_ai_competition:
+    name: install-conda-climetlab_s2s_ai_competition
+    runs-on: "ubuntu-latest"
+    defaults:
+      run:
+        shell: bash -l {0}
     steps:
     - uses: actions/checkout@v2
     - name: Setup python

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -22,7 +22,6 @@ jobs:
     - name: import climetlab_s2s_ai_competition
       run: |
         python -c "import climetlab_s2s_ai_competition"
-          conda create --name climetlab_s2s_ai_competition-dev python=3.7 xarray
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -40,6 +40,7 @@ jobs:
     - name: Set up conda environment
       run: |
           conda create --name climetlab_s2s_ai_competition-dev python=3.7 xarray
+          source activate climetlab_s2s_ai_competition-dev
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -22,5 +22,4 @@ jobs:
     - name: import climetlab_s2s_ai_competition
       run: |
         python -c "import climetlab_s2s_ai_competition"
-      run: |
         python -m pip install --upgrade pip

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -22,7 +22,6 @@ jobs:
     - name: import climetlab_s2s_ai_competition
       run: |
         python -c "import climetlab_s2s_ai_competition"
-    - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
         pip install -e .

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -47,4 +47,5 @@ jobs:
         pip install -e .
     - name: import climetlab_s2s_ai_competition
       run: |
+        which python
         python -c "import climetlab_s2s_ai_competition"

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -23,7 +23,6 @@ jobs:
       run: |
         python -c "import climetlab_s2s_ai_competition"
           conda create --name climetlab_s2s_ai_competition-dev python=3.7 xarray
-          conda activate climetlab_s2s_ai_competition-dev
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -19,6 +19,8 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install -e .
+    - name: import climetlab_s2s_ai_competition
+      run: |
         python -c "import climetlab_s2s_ai_competition"
 
   install-conda-climetlab_s2s_ai_competition:
@@ -42,4 +44,6 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install -e .
+    - name: import climetlab_s2s_ai_competition
+      run: |
         python -c "import climetlab_s2s_ai_competition"

--- a/.github/workflows/installs.yml
+++ b/.github/workflows/installs.yml
@@ -30,4 +30,3 @@ jobs:
     - name: import climetlab_s2s_ai_competition
       run: |
         which python
-        python -c "import climetlab_s2s_ai_competition"

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -21,5 +21,5 @@ jobs:
           pip install -e .
       - name: Run tests
         run: |
-          pip install pytest pytest-xdist 
-          python -m pytest -n 4 --durations=10
+          pip install pytest
+          python -m pytest --durations=10

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -22,4 +22,4 @@ jobs:
       - name: Run tests
         run: |
           pip install pytest
-          pytest
+          python -m pytest

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -21,5 +21,5 @@ jobs:
           pip install -e .
       - name: Run tests
         run: |
-          pip install pytest
+          pip install pytest pytest-xdist 
           python -m pytest

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -1,0 +1,25 @@
+name: climetlab_s2s_ai_competition testing
+
+on: pull_request
+
+jobs:
+  test:  # Runs testing suite on various python versions.
+    name: Test climetlab_s2s_ai_competition, python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    defaults:
+        run:
+          shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7]
+    steps:
+      - uses: actions/checkout@v2
+      - name:
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e .
+      - name: Run tests
+        run: |
+          pip install pytest
+          pytest

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -22,4 +22,4 @@ jobs:
       - name: Run tests
         run: |
           pip install pytest pytest-xdist 
-          python -m pytest
+          python -m pytest -n 4 --durations=10

